service/src/main/java/uk/gov/hmcts/reform/fpl/service/RespondentService.java

```diff
@@ -94,4 +98,41 @@ private String getRespondentFullName(RespondentParty respondentParty) {
 
         return String.format("%s %s", firstName, lastName);
     }
+
+    public List<RespondentSolicitor> getRegisteredSolicitors(List<Element<Respondent>> respondents) {
+        return unwrapElements(respondents)
+            .stream()
+            .filter(this::isRegisteredSolicitor)
+            .map(Respondent::getSolicitor)
+            .collect(Collectors.toUnmodifiableList());
+    }
+
+    public List<RespondentSolicitor> getUnregisteredSolicitors(List<Element<Respondent>> respondents) {
+        return unwrapElements(respondents)
+            .stream()
+            .filter(this::isUnregisteredSolicitor)
+            .map(Respondent::getSolicitor)
+            .collect(Collectors.toUnmodifiableList());
+    }
+
+    private boolean isRegisteredSolicitor(Respondent respondent) {
```

Comment:
are we checking too much? should we assume that the email is there? Pushing the logic to the Solicitor